### DISCUSSION TITLE
Idea for bugfix #2 for currency calculator

### DISCUSSION THREAD

**Discussion by SamuelTAFE** [D3nan/Tafe24S1T102_Ext#8] [General]
Hey all,

I have an idea to fix another small bug with the foreign currency calculator.

At the moment, if the user accidentally selects the same from and to currency, they get a zero answer returned (conversionRate is initialised to 0).

I tested a way to fix this in my local repo, by checking if the 
'SelectedIndex' for each combobox is the same
at the start of the program.

If it is, it throws an error, otherwise the rest of the program runs.
I've attached a screenshot for reference of how this could be fixed.

Also saw that Xiaorui has a similar idea commented out part way down the code.
This way would also work, we could use that instead of the selectedindex check I've outlined above (but same if - else structure).

Have attached screenshots for reference, what do you think?

![Bugfix2](https://github.com/D3nan/Tafe24S1T102_Ext/assets/162761329/7f79ba64-8712-4bb5-94a2-a28e5b62db42)
![Bugfix2b](https://github.com/D3nan/Tafe24S1T102_Ext/assets/162761329/bfe99aa7-f736-4922-a5ca-6c0a0445a33c)
![Bugfix2c](https://github.com/D3nan/Tafe24S1T102_Ext/assets/162761329/b9a53a9b-d5ff-462a-9f0c-bff4956233bf)

**Comment by starguo1998** [comment]
Hi Sam
Thank you for bringing up this issue and proposing potential solutions to address it. Ensuring that users cannot accidentally select the same currency in both the "from" and "to" fields is crucial for the accuracy of our foreign currency calculator.

Your proposed solution of checking the SelectedIndex for each ComboBox at the start of the program sounds promising. It provides a straightforward way to prevent users from encountering a zero conversion rate, which can be confusing and frustrating.

Xiaorui
